iree/compiler/Dialect/HAL/Target/LLVM/LinkerTool.cpp

```diff
@@ -165,6 +165,28 @@ std::string LinkerTool::findToolInEnvironment(
     }
   }
 
+  // Then search the build tree.
+  SmallString<256> mainExecutable =
+      llvm::sys::fs::getMainExecutable(nullptr, nullptr);
+  if (!mainExecutable.empty()) {
+    llvm::sys::path::remove_filename(mainExecutable);
+
+    for (auto toolName : toolNames) {
+      std::string normalizedToolName = normalizeToolNameForPlatform(toolName);
+
+      SmallString<256> pathStorage;
+      llvm::sys::path::append(pathStorage, std::string(mainExecutable),
+                              "../../third_party/llvm-project/llvm/bin/",
+                              normalizedToolName);
```

Comment:
Ok, this path searching in the build directory _seems_ to work on my Windows machine at least... but it isn't pretty :P

Reply:
(Cleaning this up then I'll send for review again)